src/configuration.c

```diff
@@ -317,11 +317,11 @@ static void set_default_plugin(struct mptcpd_config *config,
  * @param[in]     plugins Comma separated list of plugins to load.
  */
 static void set_plugins_to_load(struct mptcpd_config *config,
-                                char *plugins)
+                                char const *plugins)
 {
         config->plugins_to_load = l_queue_new();
 
-        char *token = strtok(plugins, ",");
+        char *token = strtok((char *) plugins, ",");
```

Comment:
`strtok()` modifies the string, `plugins` in this case, so this code isn't `const`-correct.  Is there a good reason to pass `plugins` as `const` here?

Reply:
The `plugins` argument is also leaked since it is a dynamically allocated string in both config file parse and command line option parse cases.  Since you don't need it after tokenization of the plugin list call `l_free()` on it at an appropriate location, perhaps prior to exiting this function.

Reply:
> `strtok()` modifies the string, `plugins` in this case, so this code isn't `const`-correct. Is there a good reason to pass `plugins` as `const` here?


There isn't actually a good reason, it is just to be consistent with the others set functions.


> The `plugins` argument is also leaked since it is a dynamically allocated string in both config file parse and command line option parse cases. Since you don't need it after tokenization of the plugin list, call `l_free()` on it at an appropriate location, perhaps prior to exiting this function.


Well, it actually can't be freed how it is currently implemented, since `strtok()` uses the string it receives, without duplicating it, so if `l_free()` was called, `mptcpd` would fail to load plugins. However, instead of saving directly the tokens, I can duplicate them, this way `l_free()` can be called prior to exiting the function and also be used when destroying the queue.

Reply:
Since the string is modified it should not be declared `const`.  No need for a cast once the `const` qualifier is removed.

```suggestion
        char *token = strtok(plugins, ",");
```